aptos-move/block-executor/src/view.rs

```diff
@@ -1001,22 +1387,25 @@ impl<'a, T: Transaction, S: TStateView<Key = T::Key>, X: Executable> TResourceVi
         state_key: &Self::Key,
         maybe_layout: Option<&Self::Layout>,
     ) -> anyhow::Result<Option<StateValue>> {
-        self.get_resource_state_value_impl(state_key, maybe_layout, ReadKind::Value)
-            .map(|res| {
-                if let ReadResult::Value(v, _layout) = res {
-                    v
-                } else {
-                    unreachable!("Read result must be Value kind")
-                }
-            })
+        self.get_resource_state_value_impl(
+            state_key,
+            UnknownOrLayout::Known(maybe_layout),
+            ReadKind::Value,
+        )
+        .map(|res| {
+            if let ReadResult::Value(v, _layout) = res {
```

Comment:
maybe we can have into_value implemented for ReadResult that internally does this.

Reply:
Done.